drivers/wifi/nrf_wifi/Kconfig.nrfwifi

```diff
@@ -40,15 +40,16 @@ config NRF70_QSPI_LOW_POWER
 
 choice NRF70_OPER_MODES
 	bool "nRF70 operating modes"
-	default NRF70_SYSTEM_MODE if !WIFI_NRF7000
-	default NRF70_SCAN_ONLY if WIFI_NRF7000
+	default NRF70_SYSTEM_MODE if !WIFI_USAGE_MODE_SCAN_ONLY
```

Comment:
we still need a link b/w nRF7000 and WIFI_USAGE_MODE_SCAN_ONLY

Reply:
This was written with the intent that `WIFI_USAGE_MODE` is more of a "hint" than a hard request. In that case, if `WIFI_USAGE_MODE_AP` is set on a nRF7000, the `NRF70_OPER_MODE` will fall through `NRF70_SYSTEM_MODE` because it isn't supported and enable `NRF70_SCAN_ONLY`.

If the consensus is for more explicit configuration failures, I can take a look at that.

Reply:
Yeah it works fine, but I don't see any Kconfig warnings with `west build -p -b nrf9160dk/nrf9160/ns  samples/net/wifi/shell --shield nrf7002ek_nrf7000 --cmake-only` it silently falls back to scan-only.

Reply:
As I said, its currently a hint, not a hard request. Do you think it should fail to build if the driver can't support the requested mode?

Reply:
At least a Kconfig warning would be good to keep the users informed.